tests/unit/test_credentials.py

```diff
@@ -0,0 +1,58 @@
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+""" Unit tests for the credentials module. """
+
+import unittest
+
+from libcodechecker.libclient import credential_manager
+
+
+class credentialsTest(unittest.TestCase):
+    """
+    Testing the credentials.
+    """
+
+    def test_get_auth_string(self):
+        """
+        Get authentication string from the credentials.
+        """
+        credentials = {
+            "client_autologin": True,
+            "credentials": {
+                "*": "global:passphrase",
+                "*:8080": "webserver:1234",
+                "localhost": "local:admin",
+                "https://localhost/Test": "local1:admin1",
+                "localhost:6251": "super:secret",
+                "https://myserver.com/Test": "serveruser:serverpassphrase"
+                       }
```

Comment:
Please add a test case that manually specifies every possible part of the url too: `https://some.server.com:12345/Product`.